client/components/survey-container/components/question-single-selection.tsx

```diff
@@ -1,17 +1,56 @@
-import { RadioControl } from '@wordpress/components';
 import { QuestionSelectionType } from './question-step';
 
 const QuestionSingleSelection = ( { onChange, question, value }: QuestionSelectionType ) => {
-	// Placeholder component
 	return (
-		<div className="question__single-selection">
-			<RadioControl
-				options={ question.options }
-				selected={ value[ 0 ] }
-				onChange={ ( value ) => {
-					onChange( question.key, [ value ] );
-				} }
-			/>
+		<div className="question-options__container" role="radiogroup">
+			{ question.options.map( ( option, index ) => {
+				const isSelected = value.includes( option.value );
+				const handleKeyDown = ( event: React.KeyboardEvent< HTMLDivElement > ) => {
+					if ( event.key === 'Enter' || event.key === ' ' ) {
+						onChange( question.key, [ option.value ] );
+					}
+				};
+
+				return (
+					<div
+						key={ index }
+						className={ `question-options__option-control components-radio-control__option ${
+							isSelected ? 'checked' : ''
+						}` }
+						role="radio"
+						tabIndex={ 0 }
+						onClick={ () => onChange( question.key, [ option.value ] ) }
+						onKeyDown={ handleKeyDown }
+						aria-checked={ isSelected ? 'true' : 'false' }
+						aria-labelledby={ `optionLabel-${ option.value } optionHelpText-${ option.value }` }
+					>
+						<input
+							type="radio"
+							id={ `option-${ option.value }` }
+							name={ question.key }
+							value={ option.value }
+							onChange={ () => onChange( question.key, [ option.value ] ) }
+							checked={ isSelected }
+							className="form-radio"
+							tabIndex={ -1 }
+							aria-hidden="true"
+						/>
+						<div className="question-options__option-label">
+							<label id={ `optionLabel-${ option.value }` } htmlFor={ `option-${ option.value }` }>
```

Comment:
To maintain the naming cohesion, I would put `optionLabel` as `option-label`.

Reply:
Good point! I will make the changes.

Reply:
This is now resolved as well.